src/classes/CON_CurrentGradeLevel_TDTM.cls

```diff
@@ -0,0 +1,136 @@
+
+        return dmlWrapper;
+    }
+    /*******************************************************************************************************
+    * @description This method creates Grade Enrollment records whenever a Contact with a Current Grade Level
+    * is inserted OR Contact's Current Grade Level is changed.
+    * @param allIdsToContacts A Map of Id to all new Contact Records
+    * @param allIdsToContactsOld A Map of Id to all old Contact Records
+    * @param dmlWrapper
+    ********************************************************************************************************/
+    private void createGradeEnrollments(Map<Id, Contact> allIdsToContacts, Map<Id, Contact> allIdsToContactsOld, dmlWrapper dmlWrapper) {
+        List<Grade_Enrollment__c> gradeEnrollmentsToInsert = new List<Grade_Enrollment__c>();
+        for (Contact eachContact : [SELECT Id, Current_Grade_Level__c FROM Contact WHERE (((Current_Grade_Level__c != NULL) AND (Current_Grade_Level__c != 'NotApplicable')
+                                                                                         AND (Current_Grade_Level__c != '--None--') AND (Current_Grade_Level__c != ''))
+                                                                                         AND Id IN :allIdsToContacts.keySet())]) {
+            if (eachContact != NULL && ((allIdsToContactsOld == NULL) || (allIdsToContactsOld != NULL && allIdsToContactsOld.get(eachContact.Id) != NULL
+                    && eachContact.Current_Grade_Level__c != allIdsToContactsOld.get(eachContact.Id).Current_Grade_Level__c))) {
+                Grade_Enrollment__c gradeEnrollment = new Grade_Enrollment__c();
+                gradeEnrollment.Contact__c = eachContact.Id;
+                gradeEnrollment.Status__c = 'Current';
+                gradeEnrollment.Grade_Level__c = eachContact.Current_Grade_Level__c;
+                gradeEnrollmentsToInsert.add(gradeEnrollment);
+            }
+        }
+        if (gradeEnrollmentsToInsert != NULL && gradeEnrollmentsToInsert.size() > 0) {
+            dmlWrapper.objectsToInsert.addAll((List<SObject>)gradeEnrollmentsToInsert);
+        }
+    }
+    /*******************************************************************************************************
+    * @description This method updates all the Grade Enrollments with Status 'Current' to 'Former' when its
+    * related Contact's Current Grade Level changes to 'Not Applicable'.
+    * @param allIdsToContacts A Map of Id to all new Contact Records
+    * @param allIdsToContactsOld A Map of Id to all old Contact Records
+    * @param dmlWrapper
+    ********************************************************************************************************/
+    private void updateGradeEnrollments(Map<Id, Contact> allIdsToContacts, Map<Id, Contact> allIdsToContactsOld, dmlWrapper dmlWrapper) {
```

Comment:
Are we checking anywhere that the Contact's Current Grade Enrollment is getting set to blank? I was poking around and if I set the Contact's Current_Grade_Level to blank, we should probably set the Current enrollment(s) to Former , (like we do when it get's set to NotApplicable). That probably wasn't in the spec but since we are here, do you know what he level of effort would be?

Reply:
Added the scenario to line 137.